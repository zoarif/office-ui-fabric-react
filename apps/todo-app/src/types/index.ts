export * from './ITodo';
export * from './ITodoTask';
export * from './ITodoTabs';
export * from './IDataProvider';