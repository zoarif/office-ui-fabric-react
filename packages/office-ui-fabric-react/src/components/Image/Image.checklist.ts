import { ChecklistStatus } from '../../demo/ComponentStatus/ComponentStatus.types';

export const ImageStatus = {
  keyboardAccessibilitySupport: ChecklistStatus.unknown,
  markupSupport: ChecklistStatus.unknown,
  highContrastSupport: ChecklistStatus.pass,
  rtlSupport: ChecklistStatus.pass,
  testCoverage: ChecklistStatus.good
};