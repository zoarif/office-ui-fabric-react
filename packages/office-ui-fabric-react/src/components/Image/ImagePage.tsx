import * as React from 'react';
import {
  ExampleCard,
  IComponentDemoPageProps,
  ComponentPage,
  PageMarkdown,
  PropertiesTableSet
} from '@uifabric/example-app-base';
import './ImagePage.scss';
import { ImageDefaultExample } from './examples/Image.Default.Example';
import { ImageCenterExample } from './examples/Image.Center.Example';
import { ImageContainExample } from './examples/Image.Contain.Example';
import { ImageCoverExample } from './examples/Image.Cover.Example';
import { ImageNoneExample } from './examples/Image.None.Example';
import { ImageMaximizeFrameExample } from './examples/Image.MaximizeFrame.Example';
import { ComponentStatus } from '../../demo/ComponentStatus/ComponentStatus';
import { ImageStatus } from './Image.checklist';

const ImageDefaultExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Image/examples/Image.Default.Example.tsx') as string;
const ImageCenterExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Image/examples/Image.Center.Example.tsx') as string;
const ImageContainExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Image/examples/Image.Contain.Example.tsx') as string;
const ImageCoverExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Image/examples/Image.Cover.Example.tsx') as string;
const ImageNoneExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Image/examples/Image.None.Example.tsx') as string;
const ImageMaximizeFrameExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Image/examples/Image.MaximizeFrame.Example.tsx') as string;

export class ImagePage extends React.Component<IComponentDemoPageProps, {}> {
  public render(): JSX.Element {
    return (
      <ComponentPage
        title='Image'
        componentName='ImageExample'
        componentUrl='https://github.com/OfficeDev/office-ui-fabric-react/tree/master/packages/office-ui-fabric-react/src/components/Image'
        exampleCards={
          <div>
            <ExampleCard title='ImageFit: Not specified' code={ ImageDefaultExampleCode }>
              <ImageDefaultExample />
            </ExampleCard>
            <ExampleCard title='ImageFit: None' code={ ImageNoneExampleCode }>
              <ImageNoneExample />
            </ExampleCard>
            <ExampleCard title='ImageFit: Center' code={ ImageCenterExampleCode }>
              <ImageCenterExample />
            </ExampleCard>
            <ExampleCard title='ImageFit: Contain' code={ ImageContainExampleCode }>
              <ImageContainExample />
            </ExampleCard>
            <ExampleCard title='ImageFit: Cover' code={ ImageCoverExampleCode }>
              <ImageCoverExample />
            </ExampleCard>
            <ExampleCard title='Maximizing the image frame' code={ ImageMaximizeFrameExampleCode }>
              <ImageMaximizeFrameExample />
            </ExampleCard>
          </div>
        }
        allowNativeProps={ true }
        nativePropsElement={ 'img' }
        propertiesTables={
          <PropertiesTableSet
            sources={ [
              require<string>('!raw-loader!office-ui-fabric-react/src/components/Image/Image.types.ts')
            ] }
          />
        }
        overview={
          <PageMarkdown>
            { require<string>('!raw-loader!office-ui-fabric-react/src/components/Image/docs/ImageOverview.md') }
          </PageMarkdown>
        }
        bestPractices={
          <div />
        }
        dos={
          <PageMarkdown>
            { require<string>('!raw-loader!office-ui-fabric-react/src/components/Image/docs/ImageDos.md') }
          </PageMarkdown>
        }
        donts={
          <PageMarkdown>
            { require<string>('!raw-loader!office-ui-fabric-react/src/components/Image/docs/ImageDonts.md') }
          </PageMarkdown>
        }
        isHeaderVisible={ this.props.isHeaderVisible }
        componentStatus={
          <ComponentStatus
            { ...ImageStatus }
          />
        }
      />
    );
  }
}
