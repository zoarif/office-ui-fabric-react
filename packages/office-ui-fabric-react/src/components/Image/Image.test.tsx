import * as React from 'react';
import * as ReactTestUtils from 'react-dom/test-utils';
import * as renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { Image } from './Image';
import { ImageBase } from './Image.base';
import { ImageFit } from './Image.types';

/* tslint:disable:no-unused-variable */
const testImage1x1 = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAAC0lEQVQImWP4DwQACfsD/eNV8pwAAAAASUVORK5CYII=';
const brokenImage = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgFcSJAAAAC0lEQVQImWP4DwQACfsD/eNV8pwAAAAASUVORK5CYII=';

describe('Image', () => {
  beforeAll(() => {
    // Manually set image height and width since there is no DOM
    Object.defineProperty(HTMLImageElement.prototype, 'naturalHeight', { get: () => 1 });
    Object.defineProperty(HTMLImageElement.prototype, 'naturalWidth', { get: () => 1 });
  });

  it('renders Image correctly', () => {
    const component = renderer.create(<Image src={ testImage1x1 } />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an image', (done) => {
    const component = ReactTestUtils.renderIntoDocument(
      <ImageBase
        src={ testImage1x1 }
        // tslint:disable-next-line:jsx-no-lambda
        onLoad={ () => done() }
      />
    );

    const image = ReactTestUtils.findRenderedDOMComponentWithTag(component as any, 'img');
    ReactTestUtils.Simulate.load(image);

  });

  it('can cover a portrait (tall) frame with a square image', () => {
    const component = mount(
      <div>
        <Image
          src={ testImage1x1 }
          width={ 1 }
          height={ 3 }
          imageFit={ ImageFit.cover }
          className='is-portraitFrame'
        />
      </div>
    );

    component.find('img').simulate('load');
    expect(component.find('.ms-Image-image--landscape')).toHaveLength(1);
  });

  it('can cover a landscape (wide) frame with a square image', () => {
    const component = mount(
      <div>
        <Image
          src={ testImage1x1 }
          width={ 3 }
          height={ 1 }
          imageFit={ ImageFit.cover }
          className='is-landscapeFrame'

        />
      </div>
    );
    component.find('img').simulate('load');
    expect(component.find('.ms-Image-image--portrait')).toHaveLength(1);
  });

  it('can cover a landscape (wide) parent element with a square image', () => {
    const component = mount(
      <div style={ { width: '10px', height: '20px' } }>
        <Image
          className='is-frameMaximizedPortrait'
          imageFit={ ImageFit.cover }
          maximizeFrame
          src={ testImage1x1 }
        />
      </div>);

    // Manually set client height and width since there is no DOM
    Object.defineProperty(HTMLDivElement.prototype, 'clientHeight', { get: () => 10, configurable: true });
    Object.defineProperty(HTMLDivElement.prototype, 'clientWidth', { get: () => 20, configurable: true });
    component.find('img').simulate('load');

    expect(component.update().find('.ms-Image-image--portrait')).toHaveLength(1);
  });

  it('can cover a portrait (tall) parent element with a square image', () => {
    const component = mount(
      <div style={ { width: '10px', height: '20px' } }>
        <Image
          src={ testImage1x1 }
          imageFit={ ImageFit.cover }
          className='is-frameMaximizedLandscape'
          maximizeFrame
        />
      </div>
    );

    // Manually set client height and width since there is no DOM
    Object.defineProperty(HTMLDivElement.prototype, 'clientHeight', { get: () => 20, configurable: true });
    Object.defineProperty(HTMLDivElement.prototype, 'clientWidth', { get: () => 10, configurable: true });

    component.find('img').simulate('load');
    expect(component.update().find('.ms-Image-image--landscape')).toHaveLength(1);
  });

  it('allows onError events to be attached', (done) => {
    const component = ReactTestUtils.renderIntoDocument(
      <ImageBase
        src={ brokenImage }
        // tslint:disable-next-line:jsx-no-lambda
        onError={ () => done() }
      />
    );

    const img = ReactTestUtils.findRenderedDOMComponentWithTag(component as any, 'img');
    ReactTestUtils.Simulate.error(img);
  });
});
