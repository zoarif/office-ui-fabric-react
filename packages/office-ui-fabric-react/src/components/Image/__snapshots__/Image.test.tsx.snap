// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders Image correctly 1`] = `
<div
  className=
      ms-Image
      {
        overflow: hidden;
      }
  style={
    Object {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <img
    alt={undefined}
    className=
        ms-Image-image
        ms-Image-image--portrait
        is-notLoaded
        is-fadeIn
        {
          display: block;
          opacity: 0;
        }
    onError={[Function]}
    onLoad={[Function]}
    role={undefined}
    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAAC0lEQVQImWP4DwQACfsD/eNV8pwAAAAASUVORK5CYII="
  />
</div>
`;
